module/data/item/__core.json

```diff
@@ -0,0 +1,30 @@
+{
+	"item": [
+		{
+			"name": "Sentinel Shield",
+			"source": "DMG",
+			"effects": [
+				{
+          "changes": [
+            {
+              "key": "flags.dnd5e.initiativeAdv",
+              "mode": 0,
+              "value": "1",
+              "priority": "20"
+            },
+            {
+              "key": "flags.midi-qol.advantage.skill.prc",
+              "mode": 0,
+              "value": "1",
+              "priority": "20"
+            }
+          ],
+          "requires": {
+            "midi-qol": true,
+            "dae": true
```

Comment:
does this actually require DAE? 🤔 

Might be a larger conversation. I would rather _not_ declare a dependence on DAE where possible, since Plutonium can handle a lot of the "basic" stuff. Once the "if dependency module not enabled -> don't add this effect" stuff is wired up, superfluously depending on DAE is going to block otherwise-perfectly-useable effects from being imported

Ultimately not too worried about it, as the general background of this whole module assumes you're using MidiQOL+DAE+etc., but it's nice to serve other combos where it's "free"

Reply:
`flags.dnd5e.initiativeAdv` appears to be core. The `flags.midi-qol.advantage` flag obviously isn't provided by DAE itself, _but_ midi-qol does have DAE as a dependency so...

Just to confirm your meaning:
Plutonium contains some AEs that _aren't_ Plutonium-specific, and work off core (like [sentinel shield](https://github.com/5etools-mirror-1/5etools-mirror-1.github.io/blob/c184604b0726a90f2b54a85d5b3a8902b8920d7e/data/foundry-items.json#L75)). If we include an AE bundle here in the patcher, it overwrites the Plutonium-native AE. This means that we'll be doubling work in some places

If that's correct, then the question then becomes whether the patcher can simply add effects onto the Plutonium-native ones, rather than overwrite them entirely since, here, there _is_ an extra effect that Plutonium can't natively handle (the midi-qol flag for advantage on Perception checks)

However, there's _then_ an issue of completeness: it'll probably be easier to test and troubleshoot AEs by simply looking here, rather than checking if Plutonium handles anything and then checking back here to see what's tacked on. I would personally lean toward the patcher dataset covering absolutely everything by itself, double-work be damned, just so it's all in one place. We can always trawl the Plutonium-native AEs you've set up once we know what we're doing

Reply:
> midi-qol does have DAE as a dependency so

suppose I it's a moot point, then (even if it _is_ a funny "soft" dependency). Any gripes I have about accuracy are going to be moot, too, if/when anyone PRs things, because I certainly ain't reviewing every effect line-by-line. Forget about it; depend on DAE til the cows come home

> If we include an AE bundle here in the patcher, it overwrites the Plutonium-native AE. This means that we'll be doubling work in some places

It _does not_ overwrite Plutonium-native AE, currently, so the `flags.dnd5e.initiativeAdv` entry is redundant, as Plutonium adds it already. My idea was to add an `"override": true` to enable that functionality, since there's plenty obvious use-cases. I prefer this over committing to the "doubling work" method. 

The use case I see here is: I want to take some part of this addon data (say this `flags.dnd5e.initiativeAdv` effect) which is vanilla dnd5e, and move it into Plutonium proper--I can do that, remove the flag, and therefore resolve this tiny bit of the "doubling work"/"double maintenance" problem. In the long run, the core data will wind up cleaner because of it.

Reply:
> I certainly ain't reviewing every effect line-by-line
Might be overly optimistic but I hope to give (official non-items) a once-over sanity check. So _hopefully_ there'll be no bad data in the core files.

Regarding merging data, I'll try to favour merging and send edits to the 5eT `foundry-*.json` files, but no promises since I'm not actually 100% sure on what even is DAE-provided and what's core 😎

Reply:
> > If we include an AE bundle here in the patcher, it overwrites the Plutonium-native AE. This means that we'll be doubling work in some places
> 
> ~It _does not_ overwrite Plutonium-native AE, currently, so the `flags.dnd5e.initiativeAdv` entry is redundant, as Plutonium adds it already. My idea was to add an `"override": true` to enable that functionality, since there's plenty obvious use-cases. I prefer this over committing to the "doubling work" method.~
> 
> ~The use case I see here is: I want to take some part of this addon data (say this `flags.dnd5e.initiativeAdv` effect) which is vanilla dnd5e, and move it into Plutonium proper--I can do that, remove the flag, and therefore resolve this tiny bit of the "doubling work"/"double maintenance" problem. In the long run, the core data will wind up cleaner because of it.~
> 
> EDIT: j/k I did it the other way around; you can specify e.g. `"merge": {"effects": true}` to have it merge, otherwise the default is "override." Although I think I do prefer "merge by default, flag to clobber"thinking

This addresses a point I was going to raise (initially I assumed this was overwrite data as you can see) that there likely be cases where we need to overwrite (if the standard plutonium data needed to be removed..I don't have a real case, but could imagine one coming up). Also, if we were merge-only, I could see getting in a loop where we're constantly reacting to updates to plutonium AE data for updates. of course that means duplicating some data, but overwrite does allow the data to be reviewed standalone.

Reply:
> if we were merge-only, I could see getting in a loop where we're constantly reacting to updates to plutonium AE data for updates

Agree, that's the main concern with the "merge" idea. Hence it should be either opt-in or opt-out; for core data, which we imagine is going to be fairly stable (and so too will Plutonium's AEs/data for it) I'd hope we can tag things as "merge" and thus remove some duplication.

> > there likely be cases where we need to overwrite
> 
> And that's why we should find an example before requesting this 😏

For effects, I would bet money on it coming up. And ultimately, the code's already been written, even if it does need inverting/tweaking 😏 

> a good case for overwriting Plutonium non-AE data

Aha, I had no idea about the reaction thing. That might be one to wire in to Plutonium itself as a "best-guess;" if the reaction has some condition text of the form "when you are hit by ..." etc. and you're running Midi-QoL (or whatever module does this; will _definitely_ need to confirm that first!). Something like this which is simple, unambiguous/unopinionated, can be fairly-easily automated, and provides a blanket improvement, I'm happy to integrate directly.

...Otherwise, every single creature in the game which has a reaction may need tricking out with custom data, which sounds like a complete nightmare (and a support burden from here to infinity).

Reply:
> That might be one to wire in to Plutonium itself as a "best-guess"

Works for me, although I am also certain that there'll be cases where simple text-recognition fails (5e tends to be like that 😔). I'll confirm which module adds this, and then make sure I know exactly how Midi handles the different options before reporting back.

> every single creature in the game which has a reaction may need tricking out with custom data

It actually gets worse than this, because _in principle_ we can also make AEs for certain 'standard' actions or traits (e.g. Swallow, grappled-and-restrained-on-hit, Brute, etc.), as well as define AEs for individual creatures (e.g. a _shadow_'s Strength Drain). While I think Surge and I are both currently focusing on player-triggered effects, ultimately monster-triggered effects are also part of the full-automation pipe dream. I'm not sure how we'd handle it yet, but it's something far down the todo list to be aware of, I guess